src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override ValueTask WriteAsync(
+            ReadOnlyMemory<byte> buffer,
+            CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// We only support ValueTask based overloads for async IO.
+        /// This overload makes sure that the method throws, instead 
+        /// of using the unoptimized path.
+        /// </summary>
+        /// <param name="buffer"></param>
+        /// <param name="offset"></param>
+        /// <param name="count"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public override Task WriteAsync(
+            byte[] buffer,
+            int offset,
+            int count,
+            CancellationToken cancellationToken)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <summary>
+        /// Called explicitly by the consumers to flush the stream,
+        /// which marks the TDS packet as the last packet in the message
+        /// and sends it to the server.
+        /// </summary>
+        /// <param name="ct"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public override Task FlushAsync(CancellationToken ct)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override ValueTask<int> ReadAsync(
+            Memory<byte> buffer,
+            CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Peeks the next byte in the stream, without consuming it.
+        /// The next call to read will return the same byte but it 
+        /// will consume it.
+        /// </summary>
+        /// <param name="isAsync"></param>
+        /// <param name="ct"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public ValueTask<byte> PeekByteAsync(bool isAsync,
+            CancellationToken ct)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// A convenience method to skip the bytes in the stream,
+        /// by allowing buffer manipulation, instead of making the consumer
+        /// allocate buffers to read and discard the packets.
+        /// </summary>
+        /// <param name="skipCount"></param>
+        /// <param name="isAsync"></param>
+        /// <param name="ct"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public ValueTask SkipReadBytesAsync(int skipCount,
```

Comment:
No, usecase specificity is not going to be put in the IO layer, the docs would say that this is preferred method of skipping bytes which prevents the caller from allocating buffers for reading and skipping. 
If there are other patterns in the driver which can benefit from Skipping rather than reading and skipping, then they can continue to use this method.
Calling out PLP usecase specifically, make the method unwelcome to use in other scenarios.